common/editor_vscode.py

```diff
@@ -54,101 +70,65 @@ def configure(self):
             )
 
-        with progress.spinner(f"Configuring {self._display_name} for project {self.git.name!r}"):
+        with progress.spinner(
+            f"Configuring {self._display_name} for project {self.git.name!r}"
+        ):
             self.workspace_directory.mkdir(parents=True, exist_ok=True)
 
-            missing_files = filter(
-                lambda path: not path.is_file(),
-                [self.workspace_path, self.launch_path, self.tasks_path],
-            )
-
-            if not list(missing_files):
-                return logger.debug("VSCode config files already exist")
-
             self._create_workspace()
             self._create_launch()
             self._create_tasks()
 
             created_files = string.join_bullet(
                 [
                     f"Workspace: {self.workspace_path}",
-                    f"Debugging: {self.launch_path}",
+                    f"Launch: {self.launch_path}",
                     f"Tasks: {self.tasks_path}",
                 ],
             )
-            logger.info(f"Created VSCode config for project {self.git.name!r}\n{created_files}")
+            logger.info(
+                f"Created VSCode config for project {self.git.name!r}\n{created_files}"
+            )
+
+    def _get_rendered_template(self, template_name, **kwargs):
+        template = self.templates.get_template(template_name)
+        return template.render(kwargs)
 
     def _create_workspace(self):
         """Create a workspace file for the project."""
-        assert isinstance(self.database, LocalDatabase)
-
-        if self.workspace_path.is_file():
-            return logger.debug("Workspace file already exists")
-
-        workspace_config = {
-            "folders": [{"path": ".."}],
-            "settings": {
-                "terminal.integrated.cwd": self.path.as_posix(),
-                "python.defaultInterpreterPath": self.database.venv.python.as_posix(),
-            },
-        }
-
-        for worktree in self.database.worktrees:
-            cast(list, workspace_config["folders"]).append({"path": worktree.path.as_posix()})
-
-        console.print(json.dumps(workspace_config, indent=4), file=self.workspace_path)
+        rendered_template = self._get_rendered_template(
+            "code-workspace.jinja",
+            DB_NAME=self.database.name,
+            ODOO_PATH=self.odoo_directory_posix,
+            VENV_PATH=self.database.venv.python.as_posix(),
+            PYTHON_PATH=PythonEnv().python.as_posix(),
+            ODEV_EXE_PATH=self.database.odev.executable.readlink().as_posix(),
+            COMMA_SEPERATED_ADDON_NAMES=",".join(
+                [
+                    d.name
+                    for d in self.path.iterdir()
+                    if (
+                        d.is_dir()
+                        and not d.name.startswith(".")
+                        and d.name not in NON_MODULE_DIRECTORIES
+                    )
```

Comment:
A static method already exists for this
```suggestion
                    if OdoobinProcess.check_addons_path(d)
```

Reply:
Actually, here I am getting the comma separated list of addons that are there. That function is for checking whether directory contains odoo addons right?

Reply:
Indeed, but then you could theoretically have directories that are not modules in there so you could use a happy mix of both. Though I agree I don't think I've ever had the case...

Reply:
The point of this is to have a list of all the modules needed from your project directory. Of course, the user can come the settings of the workspace and replace it with the modules they want to work with.